pkg/operator/discover/discover.go

```diff
@@ -73,13 +73,18 @@ func (d *Discover) Start(namespace, discoverImage, securityAccount string) error
 
 func (d *Discover) createDiscoverDaemonSet(namespace, discoverImage, securityAccount string) error {
 	privileged := true
-	ds := &extensions.DaemonSet{
+	ds := &apps.DaemonSet{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: discoverDaemonsetName,
 		},
-		Spec: extensions.DaemonSetSpec{
-			UpdateStrategy: extensions.DaemonSetUpdateStrategy{
-				Type: extensions.RollingUpdateDaemonSetStrategyType,
+		Spec: apps.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": discoverDaemonsetName,
```

Comment:
why do we need the selector for the daemonset now?

Reply:
we still need to update selectors in deployments so CI can pass